src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -398,10 +423,19 @@ public void CompileMethod(MethodWithGCInfo methodCodeNodeNeedingCode)
                 if (!ShouldSkipCompilation(MethodBeingCompiled) && !MethodSignatureIsUnstable(MethodBeingCompiled.Signature, out var _))
                 {
                     MethodIL methodIL = _compilation.GetMethodIL(MethodBeingCompiled);
+
                     if (methodIL != null)
                     {
-                        CompileMethodInternal(methodCodeNodeNeedingCode, methodIL);
-                        codeGotPublished = true;
+                        if (!FunctionJustThrows(methodIL))
+                        {
+                            CompileMethodInternal(methodCodeNodeNeedingCode, methodIL);
+                            codeGotPublished = true;
+                        }
+                        else
+                        {
+                            if (logger.IsVerbose)
+                                logger.Writer.WriteLine($"Warning: Method `{MethodBeingCompiled}` was not compiled because it always throws an exception");
```

Comment:
The change looks great to me modulo the fact that we'll probably need to turn this into a localizable resource string IIRC.

Reply:
The other similar strings are not yet localized. I'd like to do that all at once, when we do it.

Reply:
OK, sounds reasonable, thanks for pointing that out.